File: 프로그래머스/2/42578. 의상/의상.py
from functools import reduce
def solution(clothes):
    clothMap = {}
    for [_, type] in clothes:
        clothMap[type] = clothMap.get(type) + 1 if type in clothMap else  1
    return reduce(lambda x,y: (x) * (y + 1) , clothMap.values(), 1) - 1 

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> map = new HashMap<>();
        for(int i = 0; i< clothes.length; i++){
            String[] clothesTuple = clothes[i];
            Integer count = map.getOrDefault(clothesTuple[1], 0);
            map.put(clothesTuple[1], count + 1);
        }
        int answer = 1;
        for(Integer val : map.values()){
            answer *= (val+1);
        }
        return answer - 1;
    }
}